hibernate-core/src/main/java/org/hibernate/metamodel/internal/MetadataContext.java

```diff
@@ -567,6 +573,25 @@ private <X> void applyVersionAttribute(MappedSuperclass mappingType, MappedSuper
 		}
 	}
 
+	private <X> void applyGenericEmbeddableProperties(
+			PersistentClass persistentClass,
+			EntityDomainType<X> entityType) {
+		final MappedSuperclass mappedSuperclass = persistentClass.getSuperMappedSuperclass();
+		if ( mappedSuperclass != null ) {
```

Comment:
I think this should be a while loop, since a mapped super class can have another mapped super class. Also, I think you should think check if the `PersistentClass` as a superclass and then go on checking for mapped superclasses there.

I would suggest the following:

```suggestion
		final MappedSuperclass mappedSuperclass = getMappedSuperclass( persistentClass );
		while ( mappedSuperclass != null ) {
```

and at the bottom `mappedSuperclass = getMappedSuperclass( mappedSuperclass )` and the additional method:

```java
private MappedSuperclass getMappedSuperclass(PersistentClass persistentClass) {
    do {
        final MappedSuperclass mappedSuperclass = persistentClass.getSuperMappedSuperclass();
        if ( mappedSuperclass != null ) {
            return mappedSuperclass;
        }
        persistentClass = persistentClass.getSuperclass();
    while (persistentClass != null);
    return null;
}

private MappedSuperclass getMappedSuperclass(MappedSuperclass mappedSuperclass) {
    return mappedSuperclass.getSuperMappedSuperclass() != null
        ? mappedSuperclass.getSuperMappedSuperclass()
        : getMappedSuperclass( mappedSuperclass.getSuperPersistentClass() );
}
```

Reply:
Done (with while instead of do-while in `getMappedSuperclass(PersistentClass persistentClass)`  because `mappedSuperclass.getSuperPersistentClass()` could be null).

PS: it would be nice if there was a way to avoid iterating through all mapped superclasses and their properties at this point, but I couldn't think of any.